Lib/test/test_email/test_policy.py

```diff
@@ -237,6 +237,12 @@ def test_adding_default_policies_preserves_default_factory(self):
                          email.policy.EmailPolicy.header_factory)
         self.assertEqual(newpolicy.__dict__, {'raise_on_defect': True})
 
+    def test_non_ascii_chars_do_not_cause_inf_loop(self):
+        policy = email.policy.default + email.policy.strict
+        msg = email.message.EmailMessage()
+        msg['Subject'] = 'ą'*100
+        policy.fold('Subject', msg['Subject'])
```

Comment:
While you are writing the test, you may want to make an assertion about the output of this.

Perhaps it will be easier to write an assertion if you set the `max_line_length` to be shorter, so something like:

```python
policy = email.policy.default.clone(max_line_length=20)
actual = policy.fold("Subject", "ą" * 21)                                                                         
self.assertEqual(actual,
                 'Subject: \n ' +
                 '=?utf-8?q?ąąąąąąą?=\n ' +
                 '=?utf-8?q?ąąąąąąą?=\n ' +
                 '=?utf-8?q?ąąąąąąą?=\n')
```

Reply:
I tested the equality following your suggestion.